pkg/cri/store/image/image.go

```diff
@@ -91,6 +92,15 @@ func (s *Store) Update(ctx context.Context, ref string) error {
 // update updates the internal cache. img == nil means that
 // the image does not exist in containerd.
 func (s *Store) update(ref string, img *Image) error {
+	// parsing ref consistently for refCache, including adding `docker.io/library` prefix
+	if !strings.HasPrefix(ref, "sha256:") {
```

Comment:
Note that _technically_ the OCI also allows other digest formats (`sha512`, `sha384`, and optionally others); 

- https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#digests
- https://github.com/opencontainers/distribution-spec/blob/f72e5010d8441a94de8b35ebbe505b827d05fee2/spec.md?plain=1#L663
- https://github.com/opencontainers/go-digest/blob/b9e02e015be61903bbee58e3fd349114fa28e0b4/sha.go#L7-L17

So possibly this should check if it's a `digested` reference

Reply:
Updated.